spec/lib/shanty_template_plugin/template_plugin_spec.rb

```diff
@@ -0,0 +1,127 @@
+require 'spec_helper'
+require 'shanty/artifact'
+require 'shanty_template_plugin/template_plugin'
+require 'yaml'
+
+# Tests for template plugin
+module ShantyTemplatePlugin
+  RSpec.describe(TemplatePlugin) do
+    include_context('graph')
+
+    def test_output(data, expected_files, outputdir = 'build')
+      setup_project(data)
+
+      expect_artifacts(full_paths(expected_files, outputdir))
+      expect_contents(full_paths(expected_files, outputdir))
+    end
+
+    def test_artifacts(data, expected_files, outputdir = 'build')
+      setup_project(data)
+
+      expect_artifacts(full_paths(expected_files, outputdir))
+    end
+
+    def setup_project(data)
+      project.add_parent(parent)
+
+      File.open(parent_artifact, 'wb') do |f|
+        f.write(data.to_yaml)
+      end
+
+      subject.on_build(project)
+    end
+
+    def expect_artifacts(expected_files)
+      expect(subject.artifacts(project).map(&:to_local_path)).to match_array(expected_files.keys)
+    end
+
+    def expect_contents(expected_files)
+      expected_files.each do |f, exp|
+        expect(File.open(f, 'rb').read.strip).to eql(exp)
+      end
+    end
+
+    def full_paths(expected_files, outputdir)
+      expected_files.each_with_object({}) do |v, acc|
+        acc[File.join(root, 'one', outputdir, v.first)] = v.last
+      end
+    end
+
+    before(:each) do
+      File.open(File.join(root, 'one', 'test.erb'), 'w') { |f| f.write('<%= data[\'nic\'] %>') }
```

Comment:
You can use `File.write` here again. General rule of thumb is that you should use it instead if:

* You only call `write` once in the block.
* You want the `w` mode, which is the default mode of `File.write`.